schemas/property.py

```diff
@@ -26,7 +26,12 @@ def validates_property_manager_ids(self, value):
 
     @validates("name")
     def validates_uniqueness_of_name(self, value):
-        if PropertyModel.find_by_name(value):
+        def _assigning_name():
+            if "name" in self.context and value == self.context["name"]:
+                return False
+            return True
+
+        if _assigning_name() and PropertyModel.find_by_name(value):
             raise ValidationError("A property with this name already exists")
```

Comment:
I know sqlalchemy isn't that friendly when it comes to building queries. I think providing the context here is a good solution, and might be the best option. Another option would be to build a query that checks if the name exists in the database for any row that is not this property. Without looking at the documentation I'm not to sure how to do that, but I know it can be done.